delegated_translator.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/filecoin-project/index-provider/metadata"
+	"github.com/filecoin-project/storetheindex/api/v0/finder/model"
+	"github.com/filecoin-shipyard/indexstar/httpserver"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/multiformats/go-multiaddr"
+)
+
+func NewDelegatedTranslator(backend findFunc) (http.Handler, error) {
+	finder := delegatedTranslator{backend}
+	m := http.NewServeMux()
+	m.HandleFunc("/providers/", finder.find)
+	return m, nil
+}
+
+type delegatedTranslator struct {
+	be findFunc
+}
+
+func (dt *delegatedTranslator) find(w http.ResponseWriter, r *http.Request) {
+	// read out / close the request body.
+	_, err := io.ReadAll(r.Body)
+	_ = r.Body.Close()
+	if err != nil {
+		log.Warnw("failed to read original request body", "err", err)
+		http.Error(w, "", http.StatusBadRequest)
+		return
+	}
+
+	h := w.Header()
+	h.Add("Access-Control-Allow-Origin", "*")
+	h.Add("Access-Control-Allow-Methods", "GET, PUT, OPTIONS")
```

Comment:
Side note: CORS will be handled by K8S ingress controller [here](https://github.com/ipni/storetheindex/blob/main/deploy/manifests/prod/us-east-2/tenant/storetheindex/indexstar/ingress.yaml#L8) as a cross cutting HTTP routing feature. See docs [here](https://github.com/kubernetes/ingress-nginx/blob/main/docs/user-guide/nginx-configuration/annotations.md#enable-cors). 

Ok to leave this in code for folks whom would be running indexstar straight up 👍